src/app/cbp/cbp-plan/cbp-plan.component.ts

```diff
@@ -192,17 +196,15 @@ export class CbpPlanComponent implements OnInit {
           finalFilterValue = (filterAppliedonLocal ? finalFilterValue : this.filteredData).filter((data: any)=> {
             if(filterValue['timeDuration'].some((r: any)=> 
             {
-              if(r === '1w' || r === '1m') {
-                // const today = dayjs()
-                // const startOfWeek = today.startOf(r === '1w'? 'week': 'month')
-
-                // // Get the end of the current week
-                // const endOfWeek = today.endOf(r === '1w'? 'week': 'month')
-                // return dayjs(data.endDate).isSameOrAfter(dayjs(startOfWeek)) && dayjs(data.endDate).isSameOrBefore(endOfWeek)
-                return dayjs(data.endDate).isSameOrAfter(dayjs(dayjs().subtract(1, 'week'))) && dayjs(data.endDate).isSameOrBefore(dayjs())
-              } else {
-                return dayjs(data.endDate).isSameOrAfter(dayjs(dayjs().subtract(r, 'month'))) && dayjs(data.endDate).isSameOrBefore(dayjs())
+              let count = Number(r.slice(0,-2))
```

Comment:
please put the check for "r" if r is available then do the stuff

Reply:
yes it is taken care the filter will start if data is available